model-optimizer/extensions/middle/UpsampleToResample.py

```diff
@@ -81,36 +81,43 @@ def replace_pattern(self, graph: Graph, match: Dict[str, Node]):
 
         if not math.isclose(height_scale, width_scale, rel_tol=1e-5):
             log.debug('Width and height scales are not equal: {} vs {} for node {}'.format(
-                width_scale, height_scale, upsample.soft_get('name')))
+                width_scale, height_scale, upsample_name))
             return
         if depth_scale is not None and not math.isclose(height_scale, depth_scale, rel_tol=1e-5):
             log.debug('Depth and height scales are not equal: {} vs {} for node {}'.format(
-                depth_scale, height_scale, upsample.soft_get('name')))
+                depth_scale, height_scale, upsample_name))
             return
 
         if 1 in upsample.in_ports() and not upsample.in_port(1).disconnected():
             upsample.in_port(1).disconnect()
 
-        shape = Shape(graph, {'name': upsample.name + '/0_port'}).create_node()
+        shape = Shape(graph, {'name': upsample_name + '/0_port'}).create_node()
 
         layout = graph.graph['layout']
-        if input_shape_rank == 4:
-            begin = Const(graph, {'value': int64_array([get_height_dim(layout, input_shape_rank)])}).create_node()
-            factor = Const(graph, {'value': np.array([height_scale, width_scale])}).create_node()
-        else:
-            begin = Const(graph, {'value': int64_array([get_depth_dim(layout, input_shape_rank)])}).create_node()
-            factor = Const(graph, {'value': np.array([depth_scale, height_scale, width_scale])}).create_node()
-        end = Const(graph, {'value': int64_array([get_width_dim(layout, input_shape_rank) + 1])}).create_node()
-
-        stride = Const(graph, {'value': int64_array([1])}).create_node()
-        ss = StridedSlice(graph, {'name': upsample.name + '/ss_0_port',
+        ss = StridedSlice(graph, {'name': upsample_name + '/ss_0_port',
```

Comment:
You can use `create_op_with_const_inputs ` to create SS node. In the lines 106-115 just prepare the values for `begin` and `factor` (not `Node`s).

Reply:
Done